x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/slm/SlmHealthIndicatorService.java

```diff
@@ -87,25 +109,92 @@ public HealthIndicatorResult calculate(boolean explain) {
+                    "Encountered [" + unhealthyPolicies.size() + "] unhealthy snapshot lifecycle management policies.",
+                    createDetails(explain, unhealthyPolicies, slmMetadata),
+                    impacts,
+                    List.of(
+                        new UserAction(
+                            ACTION_CHECK_RECENTLY_FAILED_SNAPSHOTS,
+                            unhealthyPolicies.stream().map(SnapshotLifecyclePolicyMetadata::getName).toList()
+                        )
+                    )
+                );
+            }
+
             return createIndicator(
                 GREEN,
                 "Snapshot Lifecycle Management is running",
-                createDetails(explain, slmMetadata),
+                createDetails(explain, Collections.emptyList(), slmMetadata),
                 Collections.emptyList(),
                 Collections.emptyList()
             );
         }
     }
 
-    private static HealthIndicatorDetails createDetails(boolean explain, SnapshotLifecycleMetadata metadata) {
+    static boolean snapshotFailuresExceedWarningCount(long failedSnapshotWarnThreshold, SnapshotLifecyclePolicyMetadata policyMetadata) {
+        SnapshotInvocationRecord lastFailure = policyMetadata.getLastFailure();
+        if (lastFailure == null) {
+            // No failures yet to act on
+            return false;
+        }
+
+        // Determine if the most recent snapshot is a failure
+        SnapshotInvocationRecord lastSuccess = policyMetadata.getLastSuccess();
+        if (lastSuccess != null && lastSuccess.getSnapshotStartTimestamp() > lastFailure.getSnapshotStartTimestamp()) {
+            // Success was more recent than last failure
+            return false;
+        }
+
+        return policyMetadata.getInvocationsSinceLastSuccess() >= failedSnapshotWarnThreshold;
+    }
+
+    private static HealthIndicatorDetails createDetails(
+        boolean explain,
+        Collection<SnapshotLifecyclePolicyMetadata> unhealthyPolicies,
+        SnapshotLifecycleMetadata metadata
+    ) {
         if (explain) {
+            Map<String, Long> unhealthyPolicyFailureCounts = unhealthyPolicies.stream()
+                .collect(
+                    Collectors.toMap(
+                        SnapshotLifecyclePolicyMetadata::getName,
+                        SnapshotLifecyclePolicyMetadata::getInvocationsSinceLastSuccess
+                    )
+                );
             return new SimpleHealthIndicatorDetails(
-                Map.of("slm_status", metadata.getOperationMode(), "policies", metadata.getSnapshotConfigurations().size())
+                Map.of(
+                    "slm_status",
+                    metadata.getOperationMode(),
+                    "policies",
+                    metadata.getSnapshotConfigurations().size(),
+                    "unhealthy_policies",
+                    unhealthyPolicies.size(),
+                    "invocations_since_last_success",
```

Comment:
This is really `invocations_since_last_success_for_unhealthy_policies`. But that's kind of a mouthful. Should we maybe nest this under `unhealthy_policies`? Or only show this field if `unhealthyPolicies.size() > 0`. It might confuse users as-is.
Otherwise LGTM

Reply:
Maybe nesting the invocation counts under an unhealthy policy object is the better route. I'll update for that.